src/org/rascalmpl/uri/URIResolverRegistry.java

```diff
@@ -263,75 +320,84 @@ public ISourceLocation logicalToPhysical(ISourceLocation loc) throws IOException
 		return result;
 	}
 
-	private ISourceLocation physicalLocation(ISourceLocation loc) throws IOException {
-		while (logicalResolvers.containsKey(loc.getScheme())) {
-			Map<String, ILogicalSourceLocationResolver> map = logicalResolvers.get(loc.getScheme());
-			String auth = loc.hasAuthority() ? loc.getAuthority() : "";
-			ILogicalSourceLocationResolver resolver = map.get(auth);
-			ISourceLocation prev = loc;
-			boolean removedOffset = false;
+	private static ISourceLocation resolveAndFixOffsets(ISourceLocation loc, ILogicalSourceLocationResolver resolver, Iterable<ILogicalSourceLocationResolver> backups) throws IOException {
```

Comment:
again, moving the body of the loop to a seperate function, nothing changed. just that git can't see it.